test/setup/loadLocalDockerImage.ts

```diff
@@ -0,0 +1,36 @@
+import { SpawnSyncReturns, spawnSync } from 'child_process';
+
+// https://regex101.com/r/eK9lPd/3
+// eslint-disable-next-line security/detect-unsafe-regex
+export const dockerImageUrlRegExp = /^(?<Name>(?<=^)(?:(?<Domain>(?:(?:localhost|[\w-]+(?:\.[\w-]+)+)(?::\d+)?)|[\w]+:\d+)\/)?\/?(?<Namespace>(?:(?:[a-z0-9]+(?:(?:[._]|__|[-]*)[a-z0-9]+)*)\/)*)(?<Repo>[a-z0-9-]+))[:@]?(?<Reference>(?<=:)(?<Tag>[\w][\w.-]{0,127})|(?<=@)(?<Digest>[A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][0-9A-Fa-f]{32,}))?/;
+
+export const loadLocalDockerImage = (
+	dockerImageURL: string,
+	reload: boolean = false
+): SpawnSyncReturns<string> => {
+	try {
+		const dockerImageUrlMatch = dockerImageUrlRegExp.exec( dockerImageURL );
+
+		if ( dockerImageUrlMatch.groups.Repo ) {
+			const imageName = dockerImageUrlMatch.groups.Repo;
+			const { stdout: alreadyLoaded } = spawnSync( 'docker',
```

Comment:
This is called within the `async` `beforeServices`, should it be `sync`?

Reply:
Currently all of our Docker-related spawns are sync (e.g. `testEnv.up`, `down, etc), so it's in line with that.  I for now prefer that, but could  be easily convinced to use the async child_process `spawn` instead of `spawnSync` if I were also going to be changing all the other Docker-related spawns to async. Let me know if you have reason to think it should be async. 

For now using blocking sync operations for booting up the Docker / test environment stuff feels best to me, but marginally. I could have went either way. For now though I wouldn't be excited to factor each of those places to async methods, and then re-test everything. If we find a reason we don't like it later it wouldn't be much to amend though.

Reply:
One more note of clarification here: That there are not async things happening in `beforeServices` is understandably confusing. it is setup as an `async` method because its setup as a test hook and all our test hooks are set to async. Meanwhile our test services docker stuff (image load, up/down, start, stop) is all setup to be sync.